include/Player/MarioInit.hpp

```diff
@@ -2,471 +2,946 @@
+	    , PARAM_INIT(mDamageHeight, 110.0f)
+	    , PARAM_INIT(mAttackHeight, 50.0f)
+	    , PARAM_INIT(mTrampleRadius, 100.0f)
+	    , PARAM_INIT(mPushupRadius, 100.0f)
+	    , PARAM_INIT(mPushupHeight, 200.0f)
+	    , PARAM_INIT(mHipdropRadius, 250.0f)
+	    , PARAM_INIT(mQuakeRadius, 500.0f)
+	    , PARAM_INIT(mQuakeHeight, 500.0f)
+	    , PARAM_INIT(mTramplePowStep1, 30.0f)
+	    , PARAM_INIT(mTramplePowStep2, 40.0f)
+	    , PARAM_INIT(mTramplePowStep3, 50.0f)
+	    , PARAM_INIT(mJumpWallHeight, 1000.0f)
+	    , PARAM_INIT(mThrowPower, 2.0f)
+	    , PARAM_INIT(mSlipStart, 0.34202015f)
+	    , PARAM_INIT(mWasOnWaterSlip, 0.97f)
+	    , PARAM_INIT(mInWaterSlip, 0.89999998f)
+	    , PARAM_INIT(mToroccoRotSp, 1.0f)
+	    , PARAM_INIT(mRecoverTimer, 0xf0)
+	    , PARAM_INIT(mHotTimer, 0x4b0)
+	    , PARAM_INIT(mFeelDeep, 500.0f)
+	    , PARAM_INIT(mDamageFallHeight, 1000.0f)
+	    , PARAM_INIT(mForceSlipAngle, 0.5f)
+	    , PARAM_INIT(mClashSpeed, 50.0f)
+	    , PARAM_INIT(mHangWallMovableAngle, 0.3f)
+	    , PARAM_INIT(mColMvMax, 10.0f)
+	    , PARAM_INIT(mNoFreezeTime, 0x78)
+	    , PARAM_INIT(mKickFreezeTime, 5)
+	    , PARAM_INIT(mSurfStartFreezeTime, 0x78)
+	    , PARAM_INIT(mSleepingCheckDist, 50.0f)
+	    , PARAM_INIT(mSleepingCheckHeight, 30.0f)
+	    , PARAM_INIT(mIllegalPlaneCtInc, 4)
+	    , PARAM_INIT(mIllegalPlaneTime, 0x1e0)
+	{
+	}
+
+	TParamT<s16> mHpMax;
+	TParamT<f32> mRunningMax;
+	TParamT<f32> mDashMax;
+	TParamT<f32> mDashAcc;
+	TParamT<f32> mDashBrake;
+	TParamT<s16> mDashStartTime;
+	TParamT<s16> mWaitingRotSp;
+	TParamT<s16> mRunningRotSpMin;
+	TParamT<s16> mRunningRotSpMax;
+	TParamT<s16> mRocketRotSp;
+	TParamT<s16> mPumpingRotSpMin;
+	TParamT<s16> mPumpingRotSpMax;
+	TParamT<s16> mInvincibleTime;
+	TParamT<s16> mFootPrintTimerMax;
+	TParamT<u8> mWaterTriggerRate;
+	TParamT<u8> mGraffitoNoDmgTime;
+	TParamT<u8> mRestMax;
+	TParamT<f32> mShadowSize;
+	TParamT<f32> mShadowErase;
+	TParamT<f32> mHoldRadius;
+	TParamT<f32> mDamageRadius;
+	TParamT<f32> mDamageHeight;
+	TParamT<f32> mAttackHeight;
+	TParamT<f32> mTrampleRadius;
+	TParamT<f32> mPushupRadius;
+	TParamT<f32> mPushupHeight;
+	TParamT<f32> mHipdropRadius;
+	TParamT<f32> mQuakeRadius;
+	TParamT<f32> mQuakeHeight;
+	TParamT<f32> mTramplePowStep1;
+	TParamT<f32> mTramplePowStep2;
+	TParamT<f32> mTramplePowStep3;
+	TParamT<f32> mJumpWallHeight;
+	TParamT<f32> mThrowPower;
+	TParamT<f32> mSlipStart;
+	TParamT<f32> mWasOnWaterSlip;
+	TParamT<f32> mInWaterSlip;
+	TParamT<f32> mToroccoRotSp;
+	TParamT<s16> mRecoverTimer;
+	TParamT<s16> mHotTimer;
+	TParamT<f32> mFeelDeep;
+	TParamT<f32> mDamageFallHeight;
+	TParamT<f32> mForceSlipAngle;
+	TParamT<f32> mClashSpeed;
+	TParamT<f32> mHangWallMovableAngle;
+	TParamT<f32> mColMvMax;
+	TParamT<s16> mNoFreezeTime;
+	TParamT<s16> mKickFreezeTime;
+	TParamT<s16> mSurfStartFreezeTime;
+	TParamT<f32> mSleepingCheckDist;
+	TParamT<f32> mSleepingCheckHeight;
+	TParamT<s16> mIllegalPlaneCtInc;
+	TParamT<s16> mIllegalPlaneTime;
 };
```

Comment:
Looking at the linker map closer, all of the classes in this header are wrong, they all have to live inside of the TMario class =/
Maybe we can do it with forward declarations, but chances are that Mariol.hpp was simply a 10K LoC long file...

Reply:
Fair enough, can reorganize it in another PR :) It makes sense with what Joshua did for TMario in the header interfaces.